Fixes #9657 - merge NICs from compute profile in host create API

tstrachota opened the pull request:
- updated api docs for hosts and interfaces
- host create/update api actions now merge interfaces from compute
  profiles
- NIC type mapping extracted into a separate class
- return full host detail after host update

Commit:
Fixes #9657 - merge NICs from compute profile in host create API

- updated api docs for hosts and interfaces
- host create/update api actions now merge interfaces from compute
  profiles
- NIC type mapping extracted into a separate class
- return full host detail after host update

tstrachota commented:
@domcleal could you please tag this one as 1.8 priority?

ares commented on app/controllers/api/v2/hosts_controller.rb:
```diff
@@ -170,6 +182,32 @@ def facts
+
+      def host_attributes(params)
+        return {} if params.nil?
+
```
Could we deep_clone params here? We shouldn't change params values in this method, we should just return new hash with correctly nested values I think. The reason is that something else might expect params in their original form...

ares commented on test/unit/interface_type_mapper_test.rb:
```diff
@@ -0,0 +1,25 @@
+  def setup
+    @mapper = InterfaceTypeMapper
+  end
+
```
maybe one more test for DEFAULT_TYPE for nil?

tstrachota commented:
@ares rebased and updated according to your comments

ares commented:
@tstrachota tests are [failing](http://ci.theforeman.org/job/test_develop_pr_core/4327/database=postgresql,ruby=1.9.3/#showFailuresLink)

tstrachota commented:
@ares per the irc discussion I removed the `_destroy` check. I also added extended interfaces' rabl with `:primary, :provision` fields as they were missing.

ares commented:
ready to merge pending Mr Jenkins approval

ares commented:
@tstrachota oops, Mr. Jenkins is not happy

tstrachota commented:
@ares yes, I forgot `_delete: 1` in the tests which fails because the field name is actually `_destroy` as you've mentioned earlier today. Removed now, sorry.

ares commented:
Merged as cfd1c413362813f24bffd62c633bf6473e051740, thanks @tstrachota!
